conformance/lang/expressions/logical-expr/logical_and.balt

```diff
@@ -0,0 +1,777 @@
+    boolean _ = (a == "text" ? 2 : 1) && false; // @error static type of each expression must be a subtype of boolean
+    boolean _ = (a == "text" ? 2 : 1) && (a == "text" ? 2 : 1); // @error static type of each expression must be a subtype of boolean
+}
+
+Test-Case: error
+Description: Test static type of expression in logical expression via invalid types using checking expression.
+Labels: logical-expr, checking-expr, int:fromString
+
+function errorFunction() {
+    boolean _ = true && check int:fromString("123"); // @error static type of each expression must be a subtype of boolean
+    boolean _ = check int:fromString("123") && true; // @error static type of each expression must be a subtype of boolean
+    boolean _ = check int:fromString("123") && check int:fromString("123"); // @error static type of each expression must be a subtype of boolean
+}
+
+Test-Case: error
+Description: Test static type of expression in logical expression via invalid types using trap expression.
+Labels: logical-expr, trap-expr, multiplicative-expr
+
+function errorFunction() {
+    boolean _ = true && trap divideNum(1, 0); // @error static type of each expression must be a subtype of boolean
+    boolean _ = divideNum(1, 0) && true; // @error static type of each expression must be a subtype of boolean
+    boolean _ = divideNum(1, 0) && divideNum(1, 0); // @error static type of each expression must be a subtype of boolean
+}
+
+function divideNum(int a, int b) returns int {
+    return a / b;
+}
+
+Test-Case: error
+Description: Test static type of expression in logical expression via invalid types using xml navigate expression.
+Labels: logical-expr, xml-step-expr, xml, xml-template-expr
+
+function errorFunction() {
+    xml a = xml `<root><ele>true</ele><ele>false</ele></root>`;
+    boolean _ = true && a/*; // @error static type of each expression must be a subtype of boolean
+    boolean _ = a/* && true; // @error static type of each expression must be a subtype of boolean
+    boolean _ = a/* && a/*; // @error static type of each expression must be a subtype of boolean
+
+    boolean _ = true && a/<ele>; // @error static type of each expression must be a subtype of boolean
+    boolean _ = a/<ele> && true; // @error static type of each expression must be a subtype of boolean
+    boolean _ = a/<ele> && a/<ele>; // @error static type of each expression must be a subtype of boolean
+
+    boolean _ = true && a/<ele>/*; // @error static type of each expression must be a subtype of boolean
+    boolean _ = a/<ele>/* && true; // @error static type of each expression must be a subtype of boolean
+    boolean _ = a/<ele>/* && a/<ele>/*; // @error static type of each expression must be a subtype of boolean
+}
+
+Test-Case: output
+Description: Test static type of the result of a logical-expr.
+Labels: logical-expr, is-expr, var, equality, function-call-expr
+
+function init() {
+    var a = true && false;
+    io:println(a is boolean); // @output true
```

Comment:
These don't quite test the static type. A better test would also include trying to assign it to a boolean variable.
```suggestion
    var a = true && false;
    boolean _ = a;
    io:println(a is boolean); // @output true
```

Similarly we can have an error test where we try to assign the result to something other than a boolean.

```ballerina
var a = b1 && b2;
int _ = a; // error
```

Reply:
[1867314](https://github.com/ballerina-platform/ballerina-spec/pull/1113/commits/18673140fb7088497f2d200c5a5d370f1e005663)